node/pkg/accountant/ntt.go

```diff
@@ -0,0 +1,193 @@
+package accountant
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"go.uber.org/zap"
+)
+
+// nttEnabled returns true if NTT is enabled, false if not.
+func (acct *Accountant) nttEnabled() bool {
+	return acct.nttContract != ""
+}
+
+// nttStart initializes the NTT accountant and starts the NTT specific worker and watcher runnables.
+func (acct *Accountant) nttStart(ctx context.Context) error {
+	acct.logger.Debug("entering nttStart")
+
+	var err error
+	acct.nttDirectEmitters, acct.nttArEmitters, err = nttGetEmitters(acct.env)
+	if err != nil {
+		return fmt.Errorf("failed to set up NTT emitters: %w", err)
+	}
+
+	for emitter, enforceFlag := range acct.nttDirectEmitters {
+		tag := ""
+		if !enforceFlag {
+			tag = " in log only mode"
+		}
+		acct.logger.Info(fmt.Sprintf("will monitor%s for NTT emitter", tag), zap.Stringer("emitterChainId", emitter.emitterChainId), zap.Stringer("emitterAddr", emitter.emitterAddr))
+	}
+
+	for emitter := range acct.nttArEmitters {
+		acct.logger.Info("will monitor AR emitter for NTT", zap.Stringer("emitterChainId", emitter.emitterChainId), zap.Stringer("emitterAddr", emitter.emitterAddr))
+	}
+
+	// Start the watcher to listen to transfer events from the smart contract.
+	if acct.env == common.AccountantMock {
+		// We're not in a runnable context, so we can't use supervisor.
+		go func() {
+			_ = acct.nttWorker(ctx)
+		}()
+	} else if acct.env != common.GoTest {
+		if err := supervisor.Run(ctx, "nttacctworker", common.WrapWithScissors(acct.nttWorker, "nttacctworker")); err != nil {
+			return fmt.Errorf("failed to start NTT submit observation worker: %w", err)
+		}
+
+		if err := supervisor.Run(ctx, "nttacctwatcher", common.WrapWithScissors(acct.nttWatcher, "nttacctwatcher")); err != nil {
+			return fmt.Errorf("failed to start NTT watcher: %w", err)
+		}
+	}
+
+	return nil
+}
+
+var WH_PREFIX = []byte{0x99, 0x45, 0xFF, 0x10}
+var NTT_PREFIX = []byte{0x99, 0x4E, 0x54, 0x54}
+
+// isNTT determines if the payload bytes are for a Native Token Transfer, according to the following implementation:
+// https://github.com/wormhole-foundation/example-native-token-transfers/blob/41ac7baae5bb0f60fff2ec87603970af39dced01/test/EndpointStructs.t.sol
+func nttIsPayloadNTT(payload []byte) bool {
+	if len(payload) < 116 {
```

Comment:
What is 116?  Is that a standard NTT payload size?

Reply:
would be nice to have a constant defined for this!

Reply:
This is the offset into the message of the NTT prefix code that I need to check to see if this is a Native Token Transfer (as opposed to some other protocol message).

I added constants for these things.